src/steamship/data/plugin/prompt_generation_plugin_instance.py

```diff
@@ -0,0 +1,105 @@
+from typing import Any, Dict, Optional
+
+from steamship import Block, File, PluginInstance, SteamshipError
+from steamship.base.client import Client
+from steamship.data.plugin.plugin_instance import CreatePluginInstanceRequest
+from steamship.data.tags.tag_constants import TagKind, TagValueKey
+
+
+class PromptGenerationPluginInstance(PluginInstance):
+    """An instance of a configured prompt completion service such as GPT-3.
+
+    The `generate` method synchronously invokes the prompt against a set of variables that parameterize it.
+    The return value is a single string.
+
+    Example Usage:
+       llm = Steamship.use('prompt-generation-default', config={ "temperature": 0.9 })
+       PROMPT = "Greet {name} as if he were a {relation}."
+       greeting = llm.generate(PROMPT, {"name": "Ted", "relation": "old friend"})
+    """
+
+    def generate(self, prompt: str, variables: Optional[Dict] = None) -> str:
+        """Complete the provided prompt, interpolating any variables."""
+
+        # Interpolate the prompt with Python formatting semantics. If no variables provided, supply an empty dict.
+        try:
+            prompt_text = prompt.format(**(variables or {}))
+        except KeyError as e:
+            raise SteamshipError(
+                message="Some variables in the prompt template were not provided.", error=e
+            )
+
+        # Each prompt will be stored in Steamship as a File. The generated text will be
+        # associated with the prompt File via Steamship tags. This enables later retrieval
+        # of prompts and their results (which may be used in subsequent operations, etc.).
+        #
+        # To learn about the Steamship data model, please consult our the docs:
+        # https://docs.steamship.com/workspaces/data_model/index.html
+        file = File.create(self.client, blocks=[Block.CreateRequest(text=prompt_text)])
+
+        # This requests generation from the parameterized prompt. Tagging with our prompt generator
+        # plugin will result in a new tag that contains the generated output.
+        # We `wait()` because generation of text is done asynchronously and may take a few moments
+        # (somewhat depending on the complexity of your prompt).
+        tag_task = file.tag(plugin_instance=self.handle)
```

Comment:
If we're abstracting away seeing the file created, would it make more sense for this to use the inline tagging method?

tag_task = `self.tag(doc=prompt_text)`

Reply:
Done!